### DISCUSSION TITLE
Contributing to this repo

### DISCUSSION THREAD

**Discussion by famousfive1** [NITK-KODE/cms-api#17] [Q&A]
I would like to contribute to this repo. What is the best way to get started keeping in mind that I am not too experienced in designing backend?

**Comment by BenzeneAlcohol** [comment]
How much of backend do you know, as of now?

**Reply by famousfive1** [reply]
I have previously worked on a couple of personal projects. Know a little about interfacing with databases and responding to webpage requests (express static mostly).

**Reply by BenzeneAlcohol** [reply]
You can refer to @Utkar5hM 's comment below and start off the contributions then!

**Comment by Utkar5hM** [comment]
We are currently following strict ESLINT structure (air-bnb style) for the code as mentioned here : https://github.com/NITK-KODE/cms-api/issues/9#issuecomment-917702125

You can download eslint extension in vscode and enable its formatter for easily fixing any code structure issues or use the command line instructions mentioned by @priyansh32  from the link above.
 
we are currently working with dependencies mentioned in the  [package.json](https://github.com/NITK-KODE/cms-api/blob/main/package.json), You don't need to know all the dependencies mentioned to work on the project but only those which are necessary for the features or the part of the code you are working on.

As the API is at initial stage  of the development and a lot of refactoring and enhancements could be done including some major changes in upcoming future, There's possibility  of a lot of code to have bugs and issues, especially due to limited testing.  It will be really helpful if you can help finding those out by trying out the api which will also lead you to understand the api better and develop it.
